docs/ActiveNode.rst

```diff
@@ -200,16 +200,24 @@ By default, every ``enum`` property will be defined as ``unique``, to improve qu
       enum type: [:image, :video, :unknown], _index: false
     end
 
-Sometimes it is desirable to have a default value for an ``enum`` property.  To acheive this, you can simply define a property with the same name which defines a default value:
+Sometimes it is desirable to have a default value for an ``enum`` property.  To acheive this, you can simply pass the ``_default`` option when defining the enum:
```

Comment:
this was outdated

Reply:
You're totally right. A `_default` option is whitelisted in the source and I assumed it worked. Just testing though, I see it does not work. I'll change it back to the old documentation.

Reply:
Wait no, the `_default` option does work. There's just a bug that if someone submits an enum param that isn't among the list of options, then the record is persisted with the first option. It should either be persisted with the `_default` option or it should raise an exception. Do you have an opinion either way?

Reply:
Also, regarding `_default` vs `default`, personally, I don't really like the idea of accepting both. More code would need to be added to make sure only one option was given and it feels like clutter. Given that this is already a breaking change, now might be the time to remove `_` from all these options. 

This being said, I'm happy to do it your way if that's what you decide. It would probably involve updating `normalize_key_list` to both normalize the new keys and spit out depreciation warnings.

Reply:
So looking into persistence a bit more. Currently, if a `_default` option isn't set, and a submitted value doesn't match a permitted value, then the node will be persisted with a value of `0`. This is a **bug** if someone has defined their enums using a hash like `enum property: {one: 'ONE', two: 'TWO'}`. If you have an enum like that, and execute `Model.create(property: 'TEST')`, then the object will be persisted with `property = 0` when the only allowed values are `'ONE'` or `'TWO'`. It's a little hard to diagnose because the `Neo4jrb` object shows `property = nil` (but the db has `property = 0`).

This is what I think the fix should be: 
1. If `_default` isn't specified, and a given param doesn't match one of the values in the enum set, an exception should be raised. It could be an `ArgumentError`, but since these happen for a variety of reasons, I'm thinking a special `InvalidEnumValueError` would be better (allowing someone wants to rescue that, specifically).
2. If `_default` is specified, and a given param doesn't match one of the keys in the enum set, then the `_default` value is used as if `Model.create(property: default_value)` had been called (i.e. `default_value` will be matched against the enum keys to determine what value should be persisted to the db).

Reply:
Yeah, fair enough about supporting both.  Let's stick with the underscore for now because it's what `ActiveRecord` does (even though it's really strange...)

Really good catch about it defaulting to `0` in all cases.  That's pretty ugly...

One thing that I thought of: if we have a `_default` we should raise an exception if that isn't in the list of `enum` options.

I like 1), though regarding 2) I think that if you specify a `_default` and you give a value that isn't in the list of enums it should raise an exception.

In the end, I think you're absolutely right.  If they give something that's not in the params list we should raise an exception.  I definitely like the idea of having an `InvalidEnumValueError` which inherits from `ArgumentError` so that users can account for this.

Just checking in really quickly (still working through the backlog of issues and PRs, don't worry ;) ).  Let me know if I missed anything here

Reply:
Will do. And no rush! I've thrown a lot at you this week.